### PR TITLE
Reverts ambrosia & cannabis into contraband (importable by crate) replaces with pretty flowers & removes them from the seed vendor

### PR DESCRIPTION
**Pull request by FantasticFwoosh**
:cl: Moonlighting Mac says
del: Nanotransen have caught on about how to tax botanists over decadent recreational drug use, in a panic all the botanists hurriedly destroyed all the ambrosia & cannabis supplies in the vendors. However they made a quick swap with some old disused starthistle seeds in the hopes Nanotren wouldn't notice the large stock difference.
add: You can now harvest starthistle for seeds, starthistles also have gained 3 mutations to use & a few starthistle seeds are now inside the contraband section of the seed vendor, discover them all!
add: Freelance supply traders have caught wind of how the tax is affecting the station & are willing to offer crates of "recreational seeds" though the contraband channel of cargo. 
imageadd: Re-enabled star thistle tray sprites & added a seed sprite for starthistle
experiment: Even with the tax in place, former botanist employees may have hidden contraband around the station rather than destroy all the stock absolutely, stay vigilant and perhaps your persistence searching will be rewarded.
/:cl:

Though it is not a complete revert of ambrosia & cannabis (which may i add are good medicinal & recreational plants in the game) the easy access to them and their summary strong mutations, is why i felt it was necessary to settle upon this compromise to simply put them slightly further out of reach and delay the state of progression to Gaia or any of the end-game cannabis strains which are very potent without some input effort.

The use of starthistle (with its summary maintained new state which can be atomised but are central to the functioning of this PR) is also to not chastise botanists completely but also keep them active. The thistles need to be cultivated first, and simply charging in and spamming mutagen puts your stock supplies at risk; a wise botanist as such would harvest & distribute the seeds before mutating them with the hopes of 1/3 mutations possible.

> Summarily this PR does not chastise pod people vaults either, given their unlimited supply of mutagen & allows them to navigate around stock changes for the seed vendor.

> The starthistle plants are also good aesthetics 🎨  & tray fillers for keeping bees traveling between trays easy since they are so disposable and no longer require a shovel to remove forcibly from trays as a annoyance.

As mentioned by my experiment statement, this PR does not actively affect maps and there may be hidden/overt sources of the confiscated plant produce or seeds around in parts of the station, effort by botanists to get these out from maint or otherwise is intentional to encourage pro-activity.

### CODE REVIEW

File: code/modules/hydroponics/grown/misc.dm [hunk 1]
```diff
@@ -1,19 +1,32 @@
...
+	growing_icon = 'icons/obj/hydroponics/growing_flowers.dmi'
 	genes = list(/datum/plant_gene/trait/plant_type/weed_hardy)
+	mutatelist = list(/obj/item/seeds/ambrosia, /obj/item/seeds/cannabis, /obj/item/seeds/harebell)
+
+/obj/item/seeds/starthistle/harvest(mob/user)
+	var/obj/machinery/hydroponics/parent = loc
+	var/seed_count = yield
+	if(prob(getYield() * 20))
+		seed_count++
+		var/output_loc = parent.Adjacent(user) ? user.loc : parent.loc
```
Comment (Jalleo) [inline code/modules/hydroponics/grown/misc.dm]: Now that I'm thinking about it why is there a colon used for this line and the other botany ones have it?

Comment (XDTM) [inline code/modules/hydroponics/grown/misc.dm]: Probably you can find something similar in the seed extractor to account for tk

Comment (FantasticFwoosh) [inline code/modules/hydroponics/grown/misc.dm]: Oh yeah, there's actually a note about that on the replica pod plant.

**Comment by ma44** [comment]
>Not just making yield not -1 or slapping in a object

**Comment by FantasticFwoosh** [comment]
There's no need for it ma44. if you read the commits you'd know that making more starthistles & planting them easy is directly related to a proposed solution to the issue.

**Comment by kevinz000** [comment]
as much as i hate botany this seems like unnecessarily fucking with them, please go fwoosh.

**Comment by FantasticFwoosh** [comment]
Im shuffling the deck in favour rather than changing the suit with immediate drastic action, nothing has actively changed, its a nerf of immediate gaia tray meta's and making a stronger hand for a un-maintained plant to have a function & the nessecary attention it needed.

> I plan to add 1 more unique plant mutation to starthistles to make it worthwhile, i was thinking of mutating it into a "Scots Thistle" as to raise the mutation chance to a quarter each which is fairly divided between 2 contraband mutations & 2 generic plan mutations that are somewhat unique (harebells) and this currently unmade scots thistle.

**Comment by PKPenguin321** [comment]
>removing weed from botanists

**Comment by FantasticFwoosh** [comment]
Weed plant giveth (which can mutate), weed plant taketh away. A fitting compromise.

**Comment by Repukan** [comment]
Just remove Gaia tbh

**Comment by FantasticFwoosh** [comment]
That's too upfront, it would have been less work on me to do. And if the PR gets shut down @Repukan by being poorly handled it'll be a looong time before you could put up another one other with similar content.

**Comment by Iamgoofball** [comment]
>removing weed

:-1:

**Comment by Iamgoofball** [comment]
@Cheridan can this PR be closed and mine be merged, this is the equivalent of dealing with a protest by nuking the protesters and stuffing their ashes into a free speech zone

**Comment by Jalleo** [comment]
@Iamgoofball you are going over the top.

**Comment by Iamgoofball** [comment]
the problem is with gaia

removing 9 plants just because 1 is bad is a shitty tactic and is quite literally overkill

**Comment by Jalleo** [comment]
Its not removing its making it harder to access them. Literally two extra steps.

**Comment by Iamgoofball** [comment]
I'm sorry, do you think I haven't been coding here for 3 years or something? I know exactly what "shove it into cargo" means and exactly how that turns out.

**Comment by FantasticFwoosh** [comment]
Oh no if you don't hack the machine or bother cargo/forage for seeds on the map...

> **_you might actually have to do your hydroponics botany work job and not slack off smoking dank all day!_**

**Comment by Jalleo** [comment]
@FantasticFwoosh I just noticed codewise you havent set it up so the new crate can appear as a part of the random maint stuff.

**Comment by FantasticFwoosh** [comment]
I assume that there's some weed on the maps in maintenance, such as explicitly the 'secret maint garden' rooms or a few loose examples of ambrosia (i can name 1 place off the top of my head in box maint you might get some but thats for me to know and you to find out)

The cargo box type is specifically for jumpstarting the process rather than any kind of specific loot but ill definitely give it consideration jalleo.

**Comment by banthebantz** [comment]
why?

**Comment by FantasticFwoosh** [comment]
Promotes inter-departmental interactions for one, you need to talk to cargo if you or go through the indignity of hacking the machine yourself & scavenging the seeds, or asking a greyshirt to do it.

> I couldn't remove those plants from the vendor fully because of the knock on effect to vault people, so altering starthistle (& also maintaining since it had fallen into disrepair) to be a progenital ancestor to the plants with multiple possible mutations makes it slower in achieving a DIRECT outcome without throwing mutagen & effort at the problem.

**Comment by banthebantz** [comment]
it's just a pointless change, any half decent botanist hacks the machine within the first minute anyway

**Comment by Iamgoofball** [comment]
# daily reminder that shoving something into cargo is not "inter-department cooperation encouraging", it's "i dont like X feature but I can't get away with removing it properly so I'll hide it instead"

**Comment by FantasticFwoosh** [comment]
Procuring the tools from where? And even then they have got to throw mutagen at the problem when there are 2 seeds (requiring time to harvest to get more seeds to plant down) with 3 (to be expanded to 4 at a later date i hope) possible RNG mutations beside risk of dying if you TRY to rush it.

> You're mr magic cargo fingers, make cargo less bad then its literally our port of call for imports, without it we'd be screwed in a crisis or supply shortage

My hands were tied on the matter of making it indirectly lead into ambrosia and cannabis without actually making it cargo/maploot mutually exclusive. It was the best solution to it.

**Comment by banthebantz** [comment]
procuring a screwdriver and a multitool is easy 
git gud

**Comment by Jalleo** [comment]
You forgot the crucial item. A crowbar @banthebantz.

**Comment by octareenroon91** [comment]
@iamgoofball did you miss that the [now-in-seed-vendor] starthistle's mutation tree would point to weed and ambrosia in this PR?

This is a pretty damn aggressive change though.

**Comment by FantasticFwoosh** [comment]
I think they were referring to the seed vendor @Jalleo but in the case of the cargo import you'd be correct for deconning & setting the cargo board to 'broad'

**Comment by Cheridan** [comment]
goofball is right as always

**Comment by FantasticFwoosh** [comment]
Time to atomise this into a a seperate starthistle PR & perhaps still be able to import that fun contraband crate.

**Pull request closed without merging**